Add Kotlin Pam files (#4796)
awsdocs/aws-doc-sdk-examples: kotlin/usecases/creating_pam/Readme.md

The following table describes the AWS Lambda functions used by this application. 

| Function        |Trigger                 | Input                            | Output                                      | Uses                                |
| -------------   | ---------------------- | ---------------------------------| --------------------------------------------| ------------------------------------|
| Upload          | APIG PUT /upload       | See following example            | See following example                       | Storage bucket                      |   | DetectLabels    | S3 PutObject jpeg      | See following example            | N/A                                         | Label table                         |
| LabelsFn        | APIG GET /labels       | N/A                              | {"labels": {"maintain": {"count": 5}}       | Storage bucket, Label table         |
| PrepareDowload  | APIG POST /download    | {"labels": ["Mountain", "Lake"]} | N/A                                         | Labels table / Working bucket       |  

	

> **Note**: The Java application logic required to build these AWS Lambda functions is located later in this document.  	

Reviewer: ```suggestion
**Note**: The Java application logic required to build these AWS Lambda functions appears later in this document.  	
```

        )
        val gson = Gson()
    }
}

```

### WorkCount class

The following Kotlin code represents the **WorkCount** class.

```java
package com.example.photo

class WorkCount {
    private var count = 0

    fun getCount(): Int {
        return count
    }

    fun setCount(count: Int) {
        this.count = count
    }
}
```

## Deploy the AWS resources

At this point, you have completed all of the application business logic required for the PAM application to work. Now you need to deploy the AWS resources, including the AWS Lambda functions and API Gateway endpoints, so that the application can work. Instead of deploying all of the resources manually by using the AWS Management Console, you can use a provided AWS CDK script. Using the CDK script makes it more efficient to deploy the resources. 

**Note**: For information about the AWS CDK, see [What is the AWS CDK](https://docs.aws.amazon.com/cdk/v2/guide/home.html).
		
For complete instuctions on how to run the supplied AWS CDK script, see [PAM CDK](https://github.com/awsdocs/aws-doc-sdk-examples/tree/main/resources/cdk/photo_asset_manager).

### Update the Lambda functions

> After you execute the AWS CDK script, the Lambda functions are created. However, you need to replace the ones installed using the AWS CDK with the ones you created by using the AWS SDK for Kotlin. To perform this task, you can run Kotlin code that automatically updates the Lambda functions using a FAT JAR created by using the Kotlin SDK.  Perform these tasks.

Reviewer: ```suggestion
After you execute the AWS CDK script, the Lambda functions are created. However, you need to replace the ones installed using the AWS CDK with the ones you created by using the AWS SDK for Kotlin. To perform this task, you can run Kotlin code that automatically updates the Lambda functions using a FAT JAR created by using the Kotlin SDK. Perform the following tasks:
```

1. Update the PhotoApplicationResources with the AWS resource names that the CDK Script created. Make sure that you update **LABELS_TABLE**, **STORAGE_BUCKET**, **WORKING_BUCKET**, and **TOPIC_ARN** values. 

> 2. Create the JAR FAR by running **ShawdowJar**, as shown in this illustration. 

Reviewer: ```suggestion
2. Create the FAT JAR by running **shadowJar**, as shown in the following illustration. 
```

    val key = "PhotoAssetRestKotlin-1.0-SNAPSHOT-all.jar"
    val bucketName = "<Update value>"
    val functionNamesLabels = "<Update value>"
    val uploadFunction = "<Update value>"
    val downloadFunction = "<Update value>"
    val s3Handle = "<Update value>"

    // Replace all four AWS Lambda functions.
    updateFunctionCode(functionNamesLabels, bucketName, key)
    updateFunctionCode( uploadFunction, bucketName, key)
    updateFunctionCode(downloadFunction, bucketName, key)
    updateFunctionCode(s3Handle, bucketName, key)
    println("You have replaced all PAM Application Lambda functions!")
}

suspend fun updateFunctionCode(functionNameVal: String?, bucketName: String?, key: String?) {
    val functionCodeRequest = UpdateFunctionCodeRequest {
        functionName = functionNameVal
        publish = true
        s3Bucket = bucketName
        s3Key = key
    }

    LambdaClient { region = "us-east-1" }.use { awsLambda ->
        val response = awsLambda.updateFunctionCode(functionCodeRequest)
        awsLambda.waitUntilFunctionUpdated {
            functionName = functionNameVal
        }
        println("The last modified value is " + response.lastModified)
    }
}

```

> Once done, you will see a message that the AWS Lambda functions have been updated with the Kotlin FAT JAR. 

Reviewer: ```suggestion
When done, you see a message that the AWS Lambda functions have been updated with the Kotlin FAT JAR. 
```
